src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -0,0 +1,75 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { INTERNAL_USE_ONLY } from '../../../common/http-request';
+import { ApproveDataOptions, PaypalCommerceScriptLoader } from '../../../payment/strategies/paypal-commerce';
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _formPoster: FormPoster,
+        private _requestSender: RequestSender
+    ) {}
+
+    initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+        const paypalOptions = options.paypalCommerce;
+
+        if (!paypalOptions || !paypalOptions.clientId) {
+            throw new InvalidArgumentError();
+        }
+
+        return this._paypalScriptLoader.loadPaypalCommerce(paypalOptions.clientId)
```

Comment:
@helga1507 should not we pass currency from the quote?

Reply:
Should. I added it 
Thank you!